fhirstore/search/searcharguments.py

```diff
@@ -0,0 +1,133 @@
+from collections import defaultdict
+import re
+
+
+def url_to_dict(url_args):
+    search_args = {key: url_args.getlist(key) for key in url_args.keys()}
+    return search_args
+
+
+def parse_comma(key, value):
+    if "," in value:
+        return "multiple", {key: value.split(",")}
+    else:
+        return key, [value]
+
+
+def pre_process_params(url_args):
+    search_args = url_to_dict(url_args)
+    processed_params = defaultdict(list)
+    if search_args == {}:
+        processed_params = {}
+
+    for key, value in search_args.items():
+        if len(value) == 1:
```

Comment:
I had left a comment about this (I think) unnecessary if/else

Reply:
Actually this is necessary to handle a certain search pattern :
`/Patient?language=FR,NL&language=EN` and `/Patient?language=FR,NL`
See [here](https://www.hl7.org/fhir/search.html#combining)